### ISSUE TITLE
Minimap scroll bar stops highlighting all the selected word occurrences

### ISSUE THREAD

**Issue by glubsy** [microsoft/vscode#101552]
Issue Type: <b>Bug</b>

The minimap scrollbar sometimes fails to highlight all occurrences of the selected word.
Restarting VSCode fixes the problem. 
This happens seemingly randomly. In the screenshot there are some highlights still displayed in the scrollbar, but sometimes there is just no highlight displayed at all (highlights in the main viewport are always fine however).
It seems to happen in C++ as well as python at least.
I suspect this has something to do with the linter (python flake8 in my case) which probably overloads something in it, no idea yet.

VS Code version: Code 1.46.1 (cd9ea6488829f560dc949a8b2fb789f3cdc05f5d, 2020-06-17T21:13:08.304Z)
OS version: Linux x64 5.7.6-arch1-1 (Arch Linux)
![2020-07-02_01-51-54b](https://user-images.githubusercontent.com/12895548/86301912-6eec6300-bc07-11ea-8fad-b1f0f282e2e8.png)
![2020-07-02_01-52-13b](https://user-images.githubusercontent.com/12895548/86301911-6e53cc80-bc07-11ea-99f9-3c8fc81a9e46.png)

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-3740QM CPU @ 2.70GHz (8 x 2366)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>protected_video_decode: unavailable_off<br>rasterization: disabled_software<br>skia_renderer: disabled_off_ok<br>video_decode: unavailable_off<br>viz_display_compositor: enabled_on<br>viz_hit_test_surface_layer: disabled_off_ok<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|1, 3, 3|
|Memory (System)|15.60GB (4.23GB free)|
|Process Argv|--no-sandbox|
|Screen Reader|no|
|VM|0%|
|DESKTOP_SESSION|undefined|
|XDG_CURRENT_DESKTOP|undefined|
|XDG_SESSION_DESKTOP|undefined|
|XDG_SESSION_TYPE|tty|
</details><details><summary>Extensions (15)</summary>

Extension|Author (truncated)|Version
---|---|---
Bookmarks|ale|11.3.1
sourcepawn-vscode|Dre|0.1.4
vscode-diff|fab|1.4.0
autoconf|mae|0.1.0
dotenv|mik|1.0.1
python|ms-|2020.6.91350
cpptools|ms-|0.28.3
java|red|0.63.0
vscodeintellicode|Vis|1.2.9
vscode-java-debug|vsc|0.26.0
vscode-java-dependency|vsc|0.10.2
vscode-java-pack|vsc|0.9.1
vscode-java-test|vsc|0.23.0
vscode-maven|vsc|0.22.0
gitblame|wad|4.0.1

</details>

**Comment by alexdima** [comment]
@glubsy This is as designed. There are two features at play here, and here is what is going on:

#### 1. The built-in selection highlighter
This feature is built-in to the editor and is programming language agnostic. You can test this easiest in a `.txt` file. Basically if some text is selected, all instances of that text are highlighted inside the editor **and on the editor overview ruler**. This can be turned off via `editor.selectionHighlight`.

#### 2. The `DocumentHighlightProvider`
This is a provider of document highlights that is registered from an extension. If such a provider is registered (which I think is the case for you, since you have installed the Python extension), then this provider is invoked when resting the cursor on a word or when a word is selected and it returns a list of **semantic occurrences* and then the text of those occurrences is highlighted inside the editor **and on the editor overview ruler**.

When having a `DocumentHighlightProvider`, the built-in selection highlighter tries to take a back-seat and leave the semantic highlight provider own the highlighting. So the semantic occurrences will render in the outline view. But all instances of the text are still highlighted inside the editor, since they can be rendered with a different colour and they are useful for guidance if `Ctrl+D`/`Cmd+D` is used.

So long story short, I think while the language server is starting up, there is no `DocumentHighlightProvider`, so the selection highlighter executes at the beginning and highlights all occurrences. Once the extension is running, it begins to own the highlights and I believe the extension is doing things correctly, i.e. it only highlights an argument inside the function it sits in.

**Comment by glubsy** [comment]
@alexdima Thanks for the insightful reply. 

> Once the extension is running, it begins to own the highlights and I believe the extension is doing things correctly, i.e. it only highlights an argument inside the function it sits in

Indeed that looks absolutely right! This is exactly what is happening: a few seconds after loading the file, something is starting in the background, then the behavior changes to what you described. 
It's troubling that it didn't happen before and only started happening after I installed the flake8 linter for python. Some setting must have been changed in the process somewhere.

Anyway, I fixed my problem by changing the Language Server from `Microsoft` to `Jedi`. ``"python.languageServer": "Jedi"``
Thanks for your time!
